_posts/pl/2024-04-05-czy-wiesz-ze-wynikiem-zapytania-w-mongo-mozesz-zasilic-bezposrednio-inna-kolekcje.md

```diff
@@ -0,0 +1,93 @@
+---
+layout:    post
+title:     "Czy wiesz, że wynikiem zapytania w mongo możesz zasilić bezpośrednio inną kolekcję?"
+date:      2024-04-05T08:00:00+01:00
+published: true
+didyouknow: true
+lang: pl
+author: pstachowiak
+image: /assets/img/posts/2024-04-05-czy-wiesz-ze-wynikiem-zapytania-w-mongo-mozesz-zasilic-bezposrednio-inna-kolekcje/leaf.webp
+tags:
+- mongodb
+- merge
+---
+
+MongoDb w wersji 4.2 udostępnił nowy etap potoku agregacji (ang. aggregation pipeline) - $merge.
+Pozwala on na zapisanie wyniku agregacji do określonej kolekcji.
+
+Etap `$merge`:
+- Może wyprowadzać dane do kolekcji w tej samej lub innej bazie danych,
+- utworzyć nową kolekcję, jeśli kolekcja jeszcze nie istnieje,
+- może merge'ować wyniki do istniejącej kolekcji.
+
+```javascript
+{ 
+    $merge: {
+        into: <collection> -or- { db: <db>, coll: <collection> },
+        on: <identifier field> -or- [ <identifier field1>, ...],
+        let: <variables>,
+        whenMatched: <replace|keepExisting|merge|fail|pipeline>,
+        whenNotMatched: <insert|discard|fail>
+    }
+}
+```
+
+Etap `$merge` zawsze musi być ostatnim etapem agregacji, możemy określić następujące parametry:
+- `into` - kolekcja do której chcemy zapisać wynik, może to być kolekcja w innej bazie np.:  `into: { db:"myDB", coll:"myOutput" }`,
+- `on` - pole lub pola, które działają jako unikalny identyfikator dokumentu. Identyfikator określa, czy dokument wynikowy pasuje do istniejącego dokumentu w kolekcji wyjściowej. Domyślnie _id, jesli inne pola to musi być  na nich unikalny indeks,
```

Comment:
hmm... tego fragmentu nie łapię " Domyślnie _id, jesli inne pola to musi być  na nich unikalny indeks,..."
Jeżeli merytorycznie jest ok, to drobne poprawki:
Domyślnie _id, jeśli inne pola, to musi być  na nich unikalny indeks